tests/crypto/test_cases/CMakeLists.txt

```diff
@@ -11,48 +11,51 @@ zephyr_linker_sources(RODATA
     ${CMAKE_CURRENT_SOURCE_DIR}/custom-rodata.ld
 )
 
-zephyr_sources_ifdef(CONFIG_MBEDTLS_SHA256_C        test_sha_256.c)
-zephyr_sources_ifdef(CONFIG_MBEDTLS_SHA512_C        test_sha_512.c)
-zephyr_sources_ifdef(CONFIG_MBEDTLS_SHA256_C        test_hmac.c)
-zephyr_sources_ifdef(CONFIG_MBEDTLS_SHA256_C        test_hkdf.c)
-zephyr_sources_ifdef(CONFIG_MBEDTLS_ECDH_C          test_ecdh.c)
-zephyr_sources_ifdef(CONFIG_MBEDTLS_ECDSA_C         test_ecdsa.c)
-zephyr_sources_ifdef(CONFIG_MBEDTLS_AES_C           test_aes_ecb.c)
-zephyr_sources_ifdef(CONFIG_MBEDTLS_AES_C           test_aes_ecb_mac.c)
-zephyr_sources_ifdef(CONFIG_MBEDTLS_AES_C           test_aead.c)
-zephyr_sources_ifdef(CONFIG_MBEDTLS_ECJPAKE_C       test_ecjpake.c)
-zephyr_sources_ifdef(CONFIG_MBEDTLS_CIPHER_MODE_CBC test_aes_cbc.c)
-zephyr_sources_ifdef(CONFIG_MBEDTLS_CIPHER_MODE_CBC test_aes_cbc_mac.c)
-zephyr_sources_ifdef(CONFIG_MBEDTLS_CIPHER_MODE_CTR test_aes_ctr.c)
+if(NOT CONFIG_CRYPTO_SKIP_HASH_TESTS)
+    zephyr_sources_ifdef(CONFIG_MBEDTLS_SHA256_C        test_sha_256.c)
+    zephyr_sources_ifdef(CONFIG_MBEDTLS_SHA512_C        test_sha_512.c)
+    zephyr_sources_ifdef(CONFIG_MBEDTLS_SHA256_C        test_hmac.c)
+    zephyr_sources_ifdef(CONFIG_MBEDTLS_SHA256_C        test_hkdf.c)
+endif()
```

Comment:
curious, why is `CONFIG_CRYPTO_SKIP_HASH_TESTS` needed and it isn't it enough to have:
```
CONFIG_MBEDTLS_SHA256_C=n
CONFIG_MBEDTLS_SHA512_C=n
```
in the overlay config ?

Reply:
Yes this would be nice, but even if I set `CONFIG_MBEDTLS_SHA256_C=n`, it is forced on.